lib/route/src/Obelisk/Route.hs

```diff
@@ -527,14 +529,15 @@ maybeToEitherEncoder = unsafeMkEncoder $ EncoderImpl
 
 maybeEncoder
   :: ( MonadError Text check
+     , Eq b
      , Show a
      , Show b
      , check ~ parse
      )
   => Encoder check parse () b
   -> Encoder check parse a b
   -> Encoder check parse (Maybe a) b
-maybeEncoder f g = shadowEncoder f g . maybeToEitherEncoder
+maybeEncoder f g = unsafeSumEncoder exhaustiveOverlapCheck f g . maybeToEitherEncoder
```

Comment:
Seems better to define shadow encoder in terms of these two then use it here so we don't have to use "unsafe"* here.